### PR TITLE
Update Keystone configuration to support authN and authZ via Keycloak

### PR DESCRIPTION
**Pull request by yankcrime**
Instead of configuring EGI Check-in as an identity provider, use StackHPC's Keycloak installation as the proxy for authentication and authorization via other services.

### CODE REVIEW

File: etc/kayobe/kolla/config/keystone/wsgi-keystone.conf [hunk 1]
```diff
@@ -20,29 +20,34 @@ TraceEnable off
...
+    OIDCOAuthClientSecret {{ secrets_alaska_keycloak_client_secret }}
 
-    <Location ~ "/v3/auth/OS-FEDERATION/websso/oidc">
-            AuthType  openid-connect
-            Require   valid-user
-    </Location>
+    <LocationMatch /v3/auth/OS-FEDERATION/identity_providers/.*?/protocols/mapped/auth>
+      AuthType openid-connect
+      Require valid-user
+      LogLevel debug
```
Comment (dougszumski) [inline etc/kayobe/kolla/config/keystone/wsgi-keystone.conf:41]: I'm assuming this doesn't generate insane amounts of output?

File: etc/kayobe/secrets.yml [hunk 2]
```diff
@@ -1,62 +1,58 @@
 $ANSIBLE_VAULT;1.1;AES256
-39333130353964326264343633643338396637383736663663313561346439333330623230383435
```
Comment (dougszumski) [inline etc/kayobe/secrets.yml:2]: I'm assuming this has been updated to add only the new secrets.

**Review by dougszumski — APPROVED** [review]
Looks sensible to me. I'll take your word on the intricacies.

File: etc/kayobe/kolla/config/keystone/wsgi-keystone.conf (hunk 1 above)
Comment (yankcrime) [reply etc/kayobe/kolla/config/keystone/wsgi-keystone.conf:41]: It's not too crazy, and it's isolated to users authenticating via an IdP.  Once I've finished messing with Keycloak I'll drop that statement anyway.

**Pull request merged**